exec/vector/src/main/codegen/templates/VariableLengthVectors.java

```diff
@@ -534,15 +534,11 @@ public void setSafe(int index, byte[] bytes) {
       assert index >= 0;
 
       final int currentOffset = offsetVector.getAccessor().get(index);
-      offsetVector.getMutator().setSafe(index + 1, currentOffset + bytes.length);
-      try {
-        data.setBytes(currentOffset, bytes, 0, bytes.length);
-      } catch (IndexOutOfBoundsException e) {
-        while (data.capacity() < currentOffset + bytes.length) {
-          reAlloc();
-        }
-        data.setBytes(currentOffset, bytes, 0, bytes.length);
+      while (data.capacity() < currentOffset + bytes.length) {
```

Comment:
If we compare this implementation to the one from, say, a year or 18 months ago, I think we're back where we started.

This would actually be a good use case for a "checkedSetBytes" method that does the bounds checks. Still, whether done here or in the underlying method, the `if` statement is needed, so no savings in using a "checked" method,

Reply:
One trick used in the row set reader code is to avoid redundant re-allocs by computing the new size and rounding to a power of two. This is handy at the start with the vector doubles from 256 to 512 to 1K and the data size is, say, 600 bytes.

However, it is not clear that such an optimization is as necessary as it once was: the team's been doing a good job at adding up-front vector allocation where it was missing, reducing the gratuitous reallocs.

Reply:
I'm on the fence on this one. The change from 18 months ago resulted in a performance improvement of a few percent for variable length columns. This change essentially undoes that and takes us back to where we were. It seems that with the recent changes to batch sizing, we are on the path to eliminating the realloc behaviour that is part of the performance bottleneck for the setSafe methods. So, perhaps, it might be ok to let this one stay.